src/vrf.c

```diff
@@ -0,0 +1,266 @@
+//
+// Created by Tuposoft Collective on 23.01.2023.
+//
+
+#include <arpa/nameser.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <resolv.h>
+#include <string.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <netdb.h>
+#include <unistd.h>
+#include <stdarg.h>
+#include "vrf.h"
+
+#ifndef FREE
+#define FREE(p) free(p); p = NULL;
+#endif
+
+struct
+VRF {
+    char *email;
+    char *local_part;
+    char *domain;
+    char *mx_record;
+    char *mx_domain;
+    bool result;
+    bool catch_all;
+};
+
+static VRF_err
+send_command(int sock, char *format, ...)
+{
+    va_list args;
+    va_start(args, format);
+    char *command;
+    if (vasprintf(&command, format, args) < 0) {
+        return VRF_ERR;
+    }
+    if (send(sock, command, strlen(command), 0) < 0) {
+        return VRF_ERR;
+    }
+    va_end(args);
+    free(command);
+
+    return VRF_OK;
+}
+
+static VRF_err
+read_response(int sock, char *buffer)
+{
+    char (*b)[SMTP_DATA_LINES_MAX_LENGTH] = (char (*)[SMTP_DATA_LINES_MAX_LENGTH]) buffer;
+    if (read(sock, *b, sizeof *b) < 0) {
+        printf("Failed to read from socket.\n");
+        return VRF_ERR;
+    }
+    if (PRINT_RESPONSE) {
```

Comment:
Is `PRINT_RESPONSE` a macro? If yes we can use `#ifdef`, so that it is done at compile-time.